lib/ignore/ignore.go

```diff
@@ -648,3 +654,40 @@ func (c *modtimeChecker) Changed() bool {
 
 	return false
 }
+
+// Replaces the old Glob libraries interface so we can keep
+// compatibility with the old code.
+type Glob interface {
+	Match(string) bool
+}
+
+// Glue code to integrate the new Glob library while keeping the
+// old interface so we can rework it as soon as dubblestar adds
+// support for compiling the patterns in advance.
+type dubblestarGlobWrapper struct {
+	pattern    string // Stores the pattern, rn uncompiled.
+	separators string // Useless atm bc dubblestar
```

Comment:
Given that `separators` are not used it can be removed, but, does this work properly on Windows?

Reply:
It should work on Windows bc we use the PathMatch function which changes the separators automatically based on the platform.

Reply:
Yes, a pattern like `foo/bar` is valid on Windows and will match `foo\bar`. Backslashes in patterns also work as path separator and are not an escape character like they are on Unixes.